src/main/java/org/broadinstitute/hellbender/engine/filters/CountingReadFilter.java

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+
+/**
+ * Wrapper/adapter for ReadFilter that counts the number of reads filtered, and provides a filter count summary.
+ *
+ * For filters with complex predicates that are composed of compound and/or operators, provides counts
+ * by level for each component predicate.
```

Comment:
Add a note to the class-level comment here that counts will reflect short-circuited evaluation of the and/or operators (ie., not all read filters in a compound predicate will necessarily be evaluated every time).